### PR TITLE
Convert Avatar Selector activity to a custom pageblock - PMT #99415

### PR DESCRIPTION
**Pull request by nikolas**
This puts the Avatar Selector in a custom pageblock so Jess can put it
wherever it occurs in the session. I was thinking of it as the first
thing participants need to complete, but that's not the case.

I'm having trouble initializing this custom pageblock in the tests -
I was hoping to get some help with this (see the comments labeled TODO).
I couldn't find any examples in meaningfulconsent or nepi.

### CODE REVIEW

File: worth2/templates/main/avatar_selector_block.html [hunk 1]
```diff
@@ -13,15 +7,13 @@
 <form class="worth-avatars" method="post">
```
Comment (sdreher) [inline worth2/templates/main/avatar_selector_block.html]: This is where some of the pain might be. page.html creates a form that is used by all blocks, and feeds into PageView.submit. This block template is setting up a nested form which is not supported. (http://www.w3.org/TR/2011/WD-html5-20110525/forms.html#the-form-element) Some browsers, like Chrome, go so far as to strip the nested form tags out of the html.

File: worth2/main/tests/test_views.py [hunk 2]
```diff
@@ -12,42 +12,52 @@
...
-        self.assertRedirects(r, reverse('avatar-selector'))
+    def test_get(self):
+        r = self.client.get(self.url)
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, 'Avatar Selector Section')
+
+        # TODO: having trouble here -- why isn't this pageblock
+        # displaying correctly in the tests?
+        #
+        # self.assertTemplateUsed(r, 'main/avatar_selector_block.html')
```
Comment (sdreher) [inline worth2/main/tests/test_views.py]: I'm pretty sure the block template fragment will not appear in the "template used" list as it is being rendered by hand. See https://github.com/ccnmtl/django-pagetree/blob/master/pagetree/templatetags/render.py.

File: worth2/main/tests/test_views.py [hunk 3]
```diff
@@ -12,42 +12,52 @@
...
+        # self.assertContains(r, 'class="worth-avatars"')
```
Comment (sdreher) [inline worth2/main/tests/test_views.py]: See my comments below re: nested forms.

File: worth2/templates/main/avatar_selector_block.html (hunk 1 above)
Comment (nikolas) [inline worth2/templates/main/avatar_selector_block.html]: Oh - thanks! I didn't think of `page.html` already providing its form tag.

**Comment by nikolas** [comment]
@sdreher I fixed my tests here, the two issues were that I wasn't initializing my custom pageblock correctly (my `block_type` was incorrect`), and also my custom pageblock wasn't implementing the `create_from_dict` method (required by the check here: https://github.com/ccnmtl/django-pagetree/blob/master/pagetree/models.py#L500). I would think this method should be on the parent `PageBlock` class.. since my `AvatarSelectorBlock` isn't doing anything special there -- I'll make a pull request for pagetree.

**Pull request merged**